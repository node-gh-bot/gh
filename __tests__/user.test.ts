/**
 * Â© 2013 Liferay, Inc. <https://liferay.com> and Node GH contributors
 * (see file: CONTRIBUTORS)
 * SPDX-License-Identifier: BSD-3-Clause
 */

import { readFileSync, unlinkSync } from 'fs'
import { join } from 'path'
import { runCmd } from './testUtils'

describe('E2E: User Module Test', () => {
    it('List Current User `gh user`', done => {
        expect(runCmd('gh user')).toMatchSnapshot()
        done()
    })

    it('Generates token end to end without 2fa', done => {
        process.env.GENERATE_NEW_TOKEN = 'true'

        // when first generating this you need to add real GitHub
        // user & password credentials that doesn't 2fa enabled
        const user = 'myuser'
        const pass = 'mypass'

        expect(runCmd(`printf "${user}\n${pass}" | gh user --login`, true)).toMatchSnapshot()

        const configPath = join(__dirname, 'auth.json')
        const config = JSON.parse(readFileSync(configPath).toString())

        expect(config.github_token).toBe('234lkj23l4kj234lkj234lkj234lkj23l4kj234l')
        expect(config.github_user).toBe(user)

        unlinkSync(configPath)

        done()
    })

    it('Logout Current User `gh user --logout`', done => {
        expect(runCmd('gh user --logout')).toMatchSnapshot()
        done()
    })

    it('Show just the current username `gh user --whoami`', done => {
        expect(runCmd('gh user --whoami')).toMatchSnapshot()
        done()
    })
})
