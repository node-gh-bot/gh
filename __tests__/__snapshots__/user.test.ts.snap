// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: User Module Test Generates token end to end without 2fa 1`] = `
"In order to use GitHub's API, you will need to login with your GitHub account.
? Enter your GitHub user [25D[25C[2K[G? Enter your GitHub user m[26D[26C[2K[G? Enter your GitHub user my[27D[27C[2K[G? Enter your GitHub user myu[28D[28C[2K[G? Enter your GitHub user myus[29D[29C[2K[G? Enter your GitHub user myuse[30D[30C[2K[G? Enter your GitHub user myuser[31D[31C[2K[G? Enter your GitHub user m[26D[26C[2K[G? Enter your GitHub user my[27D[27C[2K[G? Enter your GitHub user myp[28D[28C[2K[G? Enter your GitHub user mypa[29D[29C[2K[G? Enter your GitHub user mypas[30D[30C[2K[G? Enter your GitHub user mypass[31D[31C[2K[G? Enter your GitHub user myuser[31D[31C
? Enter your GitHub password [input is hidden] [47D[47C[2K[G? Enter your GitHub password [hidden][37D[37C
Writing GH config data: /Users/liferay/dev/_node-gh/gh/__tests__/auth.json
Authentication succeed. Token written to global config.
"
`;

exports[`E2E: User Module Test List Current User \`gh user\` 1`] = `
"You're logged in as protoEvangelion
"
`;

exports[`E2E: User Module Test Logout Current User \`gh user --logout\` 1`] = `
"Logging out of user protoEvangelion
"
`;

exports[`E2E: User Module Test Show just the current username \`gh user --whoami\` 1`] = `
"protoEvangelion
"
`;
